src/Libraries/Microsoft.Extensions.Telemetry/Logging/ExtendedLogger.cs

```diff
@@ -0,0 +1,351 @@
+
+        void HandleException(Exception exception, int indent)
+        {
+            var indentStr = new string(' ', indent);
+
+            if (sb.Length > 0)
+            {
+                _ = sb.Append(indentStr);
+                _ = sb.Append("---> ");
+            }
+
+            var trace = new StackTrace(exception, config.UseFileInfoForStackTraces).ToString();
+
+#if NETCOREAPP3_1_OR_GREATER
+            trace = trace.Replace(Environment.NewLine, Environment.NewLine + indentStr + "   ", StringComparison.Ordinal).Trim(' ');
+#else
+#pragma warning disable R9A044 // Assign array of literal values to a static field for improved performance
+            trace = trace.Replace(Environment.NewLine, Environment.NewLine + indentStr + "   ").Trim(' ');
+#pragma warning restore R9A044 // Assign array of literal values to a static field for improved performance
+#endif
+
+            _ = sb.Append(exception.GetType());
+            _ = sb.Append(": ");
+            _ = sb.AppendLine(exception.Message);
+            _ = sb.Append(indentStr);
+            _ = sb.Append(trace);
+
+            if (exception is AggregateException aggregateException)
+            {
+                foreach (var ex in aggregateException.InnerExceptions)
+                {
+                    HandleException(ex, indent + IndentAmount);
+                }
+            }
+            else if (exception.InnerException != null)
+            {
+                HandleException(exception.InnerException, indent + IndentAmount);
+            }
+        }
+    }
+
+    private void ModernPath(LogLevel logLevel, EventId eventId, LoggerMessageState msgState, Exception? exception, Func<LoggerMessageState, Exception?, string> formatter)
+    {
+        var loggers = MessageLoggers;
+        var config = _factory.Config;
+
+        List<Exception>? exceptions = null;
+
+        // redact
+        JustInTimeRedactor? jitRedactors = null;
+        for (int i = 0; i < msgState.NumClassifiedProperties; i++)
+        {
+            try
+            {
+                ref var cp = ref msgState.ClassifiedPropertyArray[i];
+                var jr = JustInTimeRedactor.Get();
+                jr.Value = cp.Value;
+                jr.Redactor = config.GetRedactor(cp.Classification);
+                jr.Next = jitRedactors;
+                jitRedactors = jr;
+
+                var index = msgState.EnsurePropertySpace(1);
+                msgState.PropertyArray[index] = new(cp.Name, jr);
+            }
+            catch (Exception ex)
+            {
+                exceptions ??= new();
+                exceptions.Add(ex);
+            }
+        }
+
+        var joiner = ModernJoiner;
+        joiner.StaticProperties = config.StaticProperties;
+        joiner.Formatter = formatter;
+        joiner.SetIncomingProperties(msgState);
+
+        // enrich
+        foreach (var enricher in config.Enrichers)
+        {
+            try
+            {
+                enricher(joiner.PropertyBag);
```

Comment:
@geeknoid what if we provide here an additional information/context to an enricher? Otherwise `IStaticLogEnricher` is not that different from `ILogEnricher`, apart from using some fancy things like `AsyncContext`.
cc @andrey-noskov

Reply:
What would the enricher do with the tstate? I'm fine with the idea of doing this, so long as we have a use case to justify it.